contracts/MechMarketplace.sol

```diff
@@ -141,19 +150,37 @@ contract MechMarketplace is IErrorsMarketplace {
         );
     }
 
-    /// @dev Registers service as a mech.
-    /// @param serviceId Service id.
-    /// @param mechFactory Mech factory address.
-    /// @return mech The created mech instance address.
-    function create(uint256 serviceId, address mechFactory, bytes memory payload) external returns (address mech) {
-        mech = IMechFactory(mechFactory).createMech(address(this), serviceRegistry, serviceId, payload);
+    /// @dev Processes payment for request delivery.
+    /// @param deliveryMech Delivery agent mech address.
+    /// @param requestId Request id.
+    function _processPayment(address deliveryMech, uint256 requestId) internal virtual {
+        // Reentrancy guard
+        if (_locked > 1) {
+            revert ReentrancyGuard();
+        }
+        _locked = 2;
 
-        // This should never be the case
-        if (mech == address(0)) {
-            revert ZeroAddress();
+        // Get request Id payment
+        uint256 payment = mapRequestIdPayments[requestId];
+
+        // Process payment
+        if (payment > 0) {
+            uint256 fee = IMechManager(mechManager).fee();
+            payment = payment - (payment * fee) / 100;
+
+            // Transfer payment
+            (bool success, ) = deliveryMech.call{value: msg.value}("");
```

Comment:
Unnecessary gas overhead no? why not make it pull rather than push. I.e. just have a mapping that's mech specific that contains their accrued fees.

Reply:
Implemented